### DISCUSSION TITLE
Auto Add Wiki Links

### DISCUSSION THREAD

**Discussion by vHanda** [GitJournal/GitJournal#792] [Feature Requests]
Maybe it would be nice, if there is a button I could press and certain words which I have marked with Wiki style links before get marked with the links.

This could also optionally be done post save.

Typing `[[` and `]]` on a mobile keyboard isn't super comfortable.

---
> Please vote on this issue with a 👍 reaction. It's one of the primary mechanisms used to prioritize issues. Please also subscribe to the issue if you're interested, as that way I can discuss possible implementations and share mockups.

**Comment by dodtsair** [comment]
Would also be nice if on clicking the link it offered to create the page

Also possibly a configuration option, maybe enable camelCase links.   I have my notes in an App PersonalWiki which does autolinks like thus but it doesn't sync with git.  Some sync with box or dropbox with some manual algorithm.   Not confident in it for the long run

**Comment by dodtsair** [comment]
Auto complete would be nice too.  I think PersonalWiki is taking all the page names and feeding them to the auto complete of the android key board.

If i create a page FiveVsFive.  Then on another page i type in Five it suggests FiveVsFive as an auto complete option

If

**Comment by dodtsair** [comment]
One more option would be to support relative links. 

If i had some text like ./FiveVsFive then it would automatically create a link to the FiveVsFive page

**Comment by vHanda** [comment]
> One more option would be to support relative links.
> 
> If i had some text like ./FiveVsFive then it would automatically create a link to the FiveVsFive page

Hello. Just to clarify you would like local link support inside a wiki link? Currently local links only work standard markdown links. Eg - `[word](../foo.md)`

**Comment by vHanda** [comment]
>
> Also possibly a configuration option, maybe enable camelCase links. 

Could you please provide an example of the camel case? I don't quite follow.

**Comment by dodtsair** [comment]
You might write `[FiveVsFive](FiveVsFive.md)`.  But if you want to do a lot of quick links that's a lot of typing and special characters and repeated characters.   Instead you look for camelCase if i type `FiveVsFive` it would result in the same think a link to a local md file of the same name

**Comment by vHanda** [comment]
> Auto complete would be nice too. I think PersonalWiki is taking all the page names and feeding them to the auto complete of the android key board.
> 
> If i create a page FiveVsFive. Then on another page i type in Five it suggests FiveVsFive as an auto complete option
> 
> If

I've created https://github.com/GitJournal/GitJournal/issues/210 to track this bug, please could you vote on it.

**Comment by vHanda** [comment]
> You might write `[FiveVsFive](FiveVsFive.md)`. But if you want to do a lot of quick links that's a lot of typing and special characters and repeated characters. Instead you look for camelCase if i type `FiveVsFive` it would result in the same think a link to a local md file of the same name

I'm sorry, I still don't follow. Could you please open a new feature request with a bit more information.

Things that are not clear to me -
* Is this a request for Wiki Links or Normal Markdown Links - Your example is of markdown links
* Do you mean after you type 'FiveVsFive' it automatically checks if there is a file with that name and converts it to `[FiveVsFive](FiveVsFive.md)`? In that case I'm not sure what camel case has to do with this.

Sorry if I'm being a bit slow.

**Comment by bergie** [comment]
One idea would be a "link" button somewhere in the Markdown editor. It could provide a "popup version" of the normal note search screen, and clicking a note would insert a link to that.

Kind of like how [Confluence does it](https://support.atlassian.com/confluence-cloud/docs/insert-links-and-anchors/#InsertLinksandAnchors-neweditor). Listing recently opened pages there immediately is also nice, as often this means you don't even need to type a search as the page you want to link to is already in the list.
